apps/full-site-editing/full-site-editing-plugin/dotcom-fse/editor/block-inserter/index.js

```diff
@@ -26,14 +29,34 @@ function renderPostContentBlockInserter() {
 			return;
 		}
 		clearInterval( editPostHeaderToolbarInception );
+		injectBlockInserter();
+
+		// Re-inject the FSE inserter as needed in case React re-renders the header
+		const wpbody = document.getElementById( 'wpbody' );
+		if ( wpbody && typeof window.MutationObserver !== 'undefined' ) {
+			const observer = new window.MutationObserver( injectBlockInserter );
```

Comment:
Clever workaround! I learned something new today, didn't know about the `MutationObserver` API.

Reply:
> Clever workaround!

Agreed!  I didn't know about the `MutationObserver` either, this is very interesting and great to see.  Nice work here. 😄